.github/workflows/pull-requests.yml

```diff
@@ -590,23 +595,138 @@ jobs:
            fi
           done
 
-      - name: Request clusters info
+      - name: Check if Global L0 node is ready
         run: |
-          sleep 5
-          echo "Dag L1"
-          curl http://127.0.0.1:9100/cluster/info
-          echo ""
-          
-          echo "Currency L0"
-          curl http://127.0.0.1:9400/cluster/info
-          echo ""
-          
-          echo "Currency L1"
-          curl http://127.0.0.1:9700/cluster/info
-          echo ""
+          for ((i = 1; i <= 12; i++)); do
+            if [[ $(curl -s http://127.0.0.1:9000/node/info | jq '.state=="Ready"') == true ]]; then
+              echo "Global L0 is ready"
+              exit 0 
+            else
+              if [ $i -eq 12 ]; then
+                echo "Global L0 is not ready, check the logs"
+                exit 1
+              fi
+              echo "Global L0 not ready yet, trying again in 10s"
+              sleep 10
+            fi
+          done
+
+      - name: Check if DAG L1 nodes are Ready
+        run: |
+          for ((i = 1; i <= 12; i++)); do
+            if [[ $(curl -s http://127.0.0.1:9100/node/info | jq '.state=="Ready"') == true && $(curl -s http://127.0.0.1:9200/node/info | jq '.state=="Ready"') == true && $(curl -s http://127.0.0.1:9300/node/info | jq '.state=="Ready"') == true ]]; then
+              echo "All DAG L1 nodes are ready"
```

Comment:
here the same. you can extract the ready check to some bash script and execute it like:
```
./node-is-ready --port 9100 && ./node-is-ready --port 9200 ...
```

Reply:
or... the best would be to write typescript scripts and execute them only in here. bash scripts are fine to some extent but at some point for more complex checks ts is just easier (or python, go, whatever)

Reply:
Done